### ISSUE TITLE
handler list wrapper

### ISSUE THREAD

**Issue by shenek** [shenek/streamson#172]
currently a lot of API calls uses simple `Vec` for storing a list of handler.

This may not be efficient in some case:
Chaining of output has to be done directly inside strategy and is not really that simple.
Some handlers won't work well when chained.
E.g. I want to store original values in a file (using file handler and perform some convert operation). File handler by default doesn't return any output. Which is good, because it can be a bit faster.

Basically there can be two kinds of handlers.
Consumers and convertors. While consumers would be read only handlers. The converters would chain.

**Comment by shenek** [comment]
related to #162
